src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.health = health || 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+  }
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    const isHerbivore = herbivore instanceof Herbivore;
+    const isVisible = !herbivore.hidden;
+
+    if (isVisible && isHerbivore) {
+      herbivore.health -= 50;
+
+      if (herbivore.health <= 0) {
+        Animal.alive = Animal.alive.filter(animal =>
+          animal.health > 0);
+      }
+    }
```

Comment:
```suggestion
    if (isVisible && isHerbivore) {
      herbivore.health -= 50;
    }
    
    if (herbivore.health <= 0) {
        Animal.alive = Animal.alive.filter(animal =>
          animal.health > 0);
    }
```
there is no need for it nested conditions

Reply:
we can avoid filtering all animals in case animal has 50hp after bite

Reply:
filtering won't work in both cases, but in case of nesting, we just won't check the second condition if the first one is false, maybe you're right, that's better)

Reply:
Don't use nested ifs when it is possible